docs/API/V1/inventory.md

```diff
@@ -0,0 +1,39 @@
+# `GET /api/v1/datasets/inventory/<dataset><path>`
+
+This API returns an `application/octet-stream` document containing the raw byte stream of a regular file (or soft link to a regular file) at the relative `<path>` within the `<dataset>` tarball representation.
+
+## URI parameters
+
+`<dataset>` string
+The resource ID of a Pbench dataset on the server.
+
+`<path>`    string
+The relative resource path of an item in the dataset inventory, as captured by the Pbench agent packaging; for example, `/metadata.log` for the dataset metadata, or `/1-default/sample1/result.txt` for the default first iteration results.
+
+## Request headers
+
+`authorization: bearer` token \
+*Bearer* schema authorization is required to access any non-public dataset. E.g., `authorization: Bearer <token>`
+
+## Response headers
+
+`content-type: application/octet-stream` \
+The return is a raw byte stream representing the contents of the named file.
+
+## Response status
+
+`401`   **UNAUTHORIZED** \
+The client did not provide an authentication token and there is no public dataset with the name `<dataset>`.
+
+`403`   **FORBIDDEN** \
+The named `<dataset>` is not public and the authenticated user lacks authorization to read it.
+
+`404`   **NOT FOUND** \
+Either the `<dataset>` or the relative `<path>` within the dataset does not exist
+
+`415` **UNSUPPORTED MEDIA TYPE** \
+The `<path>` refers to a directory. Use `/api/v1/dataset/contents/<dataset><path>` to acquire a JSON response document describing the directory contents.
+
+## Response body
+
+The raw byte stream of the regular file will be returned.
```

Comment:
Perhaps, "The contents of the file will be returned as a raw byte stream"?

We should mention that it's the contents of the file, and we should omit "regular" since we'll also handle symlinks, right?  (If we want to discuss what happens if the target is a symlink, this would be the spot to do it, I think...but I would just remain silent on the topic.)

Reply:
Technically, the proposal is to effectively hide the symlinks by reporting them as if they were files/directories (albeit with with `"type": "sym"`), and if we have a symlink in the `files` list, and you use that URL, you're getting the contents of the "regular file" to which that symlink points, *not* the contents of the symlink file.

I kinda hoped that "regular file" would bridge that gap in one concise reference rather than writing out a paragraph. Perhaps not. 😦

Reply:
"Regular" file typically refers to not-directory, not-block-special, not-char-special, not-FIFO, not-device, not-symlink, etc.  😛  So, yeah, if someone applies this API to a symlink, by that rubric we won't return the contents of the symlink...we'll presumably return the contents of its referent.  But, since we care, it would probably be worth a short paragraph pointing that out.

Btw, when we generate the URL, we're going to need to handle symlinks to symlinks....  😝

And, we need to decide what a broken symlink will return (I _think_ I prefer a `404`, but someone might wish for the contents of the symlink file itself...perhaps we can return that in the body or message).